src/ray/object_manager/common.cc

```diff
@@ -93,6 +96,38 @@ Status PlasmaObjectHeader::TryToAcquireSemaphore(
         break;
       }
     } while (std::chrono::steady_clock::now() < *timeout_point);
+
+#else  // defined(__Linux__)
+
+    // Calculate the relative time to wait
+    auto now = std::chrono::steady_clock::now();
+    if (*timeout_point <= now) {
+      // If the timeout is already passed, try once and return
+      got_sem = sem_trywait(sem) == 0;
+    } else {
+      auto duration = *timeout_point - now;
+      auto secs = std::chrono::duration_cast<std::chrono::seconds>(duration);
+      auto ns = std::chrono::duration_cast<std::chrono::nanoseconds>(duration - secs);
+
+      timespec ts;
+      clock_gettime(CLOCK_REALTIME, &ts);  // get current time
```

Comment:
can we derive `timespec` from `auto now = std::chrono::steady_clock::now();`?

Reply:
+1 

Also, should be CLOCK_MONOTONIC rather than CLOCK_REALTIME